src/main/java/com/google/cloud/tools/gradle/appengine/core/DownloadCloudSdkTask.java

```diff
@@ -33,32 +31,30 @@
 import com.google.cloud.tools.managedcloudsdk.update.SdkUpdater;
 import java.io.IOException;
 import org.gradle.api.DefaultTask;
+import org.gradle.api.GradleException;
 import org.gradle.api.tasks.TaskAction;
 
 public class DownloadCloudSdkTask extends DefaultTask {
 
-  private CloudSdkBuilderFactory cloudSdkBuilderFactory;
-  private ManagedCloudSdkFactory managedCloudSdkFactory;
+  private ManagedCloudSdk managedCloudSdk;
 
-  public void setCloudSdkBuilderFactory(CloudSdkBuilderFactory cloudSdkBuilderFactory) {
-    this.cloudSdkBuilderFactory = cloudSdkBuilderFactory;
-  }
-
-  public void setManagedCloudSdkFactory(ManagedCloudSdkFactory managedCloudSdkFactory) {
-    this.managedCloudSdkFactory = managedCloudSdkFactory;
+  public void setManagedCloudSdk(ManagedCloudSdk managedCloudSdk) {
+    this.managedCloudSdk = managedCloudSdk;
   }
 
   /** Task entrypoint : Download/update Cloud SDK. */
   @TaskAction
   public void downloadCloudSdkAction()
-      throws UnsupportedOsException, BadCloudSdkVersionException, ManagedSdkVerificationException,
-          ManagedSdkVersionMismatchException, InterruptedException, CommandExecutionException,
-          SdkInstallerException, CommandExitException, IOException {
+      throws ManagedSdkVerificationException, ManagedSdkVersionMismatchException,
+          InterruptedException, CommandExecutionException, SdkInstallerException,
+          CommandExitException, IOException {
+    if (managedCloudSdk == null) {
```

Comment:
Here too, I think we can delegate to the `@Input` annotation to ensure the property is set.